llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -306,6 +308,9 @@ void collectCompositeElementsDefaultValuesRecursive(
     size_t NumBytes = M.getDataLayout().getTypeStoreSize(C->getType());
     std::fill_n(std::back_inserter(DefaultValues), NumBytes, 0);
     Offset += NumBytes;
+    // Print tuple {Offset, Size, DefaultValue}.
+    LLVM_DEBUG(dbgs() << "{" << Offset - NumBytes << ", " << NumBytes << ", "
```

Comment:
I don't have any data about how often folks use `LLVM_DEBUG` to quickly hack some debugging messages (I personally just use plain `llvm::outs()` for that), but if we know of someone who does that, then we should probably use `DEBUG_WITH_TYPE` to allow both generic debug messages and messages which are specific to certain functionality .

With the current approach, if anyone want's to see some printouts from the pass using `LLVM_DEBUG` mechanism, they may have to filter out this output because it is irrelevant to their investigation

Reply:
I think they could easily do that by defining their own `DEBUG_TYPE` --it just takes 1 line-- and then pass `--debug-only=<custom_type>` . Also, I grepped `LLVM_DEBUG` and there are +11k occurrences, so it's definitely better to filter only their own option if they want to quickly hack some own debugging messages.

Reply:
Yeah, `DEBUG_WITH_TYPE` is handy when you want to have several different filters in a single TU. Otherwise, `DEBUG_TYPE` is the way to go.